backend/src/auto_boundaries.rs

```diff
@@ -23,10 +23,53 @@ impl MapModel {
                 }
             })
             .chain(self.railways.iter())
-            .chain(self.waterways.iter());
+            .chain(self.waterways.iter())
+            .filter(|line_string| {
+                if let Some(context_data) = self.context_data.as_ref() {
+                    context_data
+                        .settlements
+                        .relate(*line_string)
+                        .is_intersects()
+                } else {
+                    true
+                }
+            })
+            .chain({
+                if let Some(context_data) = self.context_data.as_ref() {
+                    // Include the settlement boundary as a splitter to ensure we include settlement
+                    // outskirts.
+                    //
+                    // To get an intuition, imagine a classic ring-road topology. The ring road is typically
+                    // near the edge, but not *at* the edge of a settlement. I'm calling that area
+                    // of the settlement beyond the ring road the "outskirt".
+                    //
+                    // If we were splitting boundaries only between severances, then this outskirt
+                    // will become part of a potentially huge boundary that starts from just
+                    // outside the ring road and continues until it hits another major severance,
+                    // which could be a 100 miles away.
+                    //
+                    // NOTE: This approach is an improvement, but isn't perfect. Especially with small
```

Comment:
Yeah, in all honesty, if somebody needs to work on something around the edge of a study area, they're going to have a much better time just importing a custom boundary that includes the right surrounding area. I don't think in practice this will need to happen.